tests/test_threading.py

```diff
@@ -0,0 +1,94 @@
+"""
+test threading-specific functionality
+"""
+
+from __future__ import annotations
+
+import _setup_test_env  # noqa
+from subprocess import Popen, PIPE, STDOUT, CalledProcessError
+import re
+import os
+import sys
+import unittest
+import tempfile
+import textwrap
+from returnn.util import better_exchook
+
+
+__my_dir__ = os.path.dirname(os.path.abspath(__file__))
+__base_dir__ = os.path.dirname(__my_dir__)
+__main_entry__ = __base_dir__ + "/rnn.py"
+py = sys.executable
+PY3 = sys.version_info[0] >= 3
+
+
+def build_env():
+    """build env"""
+    env_update = os.environ.copy()
+    return env_update
+
+
+def run(args, input=None):
+    """run subproc"""
+    args = list(args)
+    print("run:", args)
+    # RETURNN by default outputs on stderr, so just merge both together
+    p = Popen(args, stdout=PIPE, stderr=STDOUT, stdin=PIPE, env=build_env())
```

Comment:
I think you don't need `env`?
```suggestion
    p = Popen(args, stdout=PIPE, stderr=STDOUT, stdin=PIPE)
```